src/libasr/codegen/llvm_utils.cpp

```diff
@@ -373,6 +373,129 @@ namespace LFortran {
     }
 
+    llvm::Value* LLVMList::find_item_position(llvm::Value* list,
+                                    llvm::Value* item) {
+        llvm::Type* pos_type = std::get<2>(typecode2listtype["i64"]);
+        llvm::Value* current_end_point = LLVM::CreateLoad(*builder,
+                                        get_pointer_to_current_end_point(list));
+        llvm::AllocaInst *i = builder->CreateAlloca(pos_type, nullptr);
+        LLVM::CreateStore(*builder, llvm::ConstantInt::get(
+                                    context, llvm::APInt(32, 0)), i);
+        llvm::Value* tmp = nullptr;
+
+        /* Equivalent in C++:
+         * int i = 0;
+         * while(list[i] != item && end_point > i) {
+         *     i++;
+         * }
+         *
+         * if (i == end_point) {
+         *    std::cout << "The list does not contain the element";
+         * }
+         */
+
+        llvm::BasicBlock *loophead = llvm::BasicBlock::Create(context, "loop.head");
+        llvm::BasicBlock *loopbody = llvm::BasicBlock::Create(context, "loop.body");
+        llvm::BasicBlock *loopend = llvm::BasicBlock::Create(context, "loop.end");
+
+        // head
+        llvm_utils->start_new_block(loophead);
+        {
+            llvm::Value *cond = builder->CreateAnd(builder->CreateICmpNE(
+                read_item(list, LLVM::CreateLoad(*builder, i), false), item),
+                builder->CreateICmpSGT(current_end_point,
+                                       LLVM::CreateLoad(*builder, i)));
+            builder->CreateCondBr(cond, loopbody, loopend);
+        }
+
+        // body
+        llvm_utils->start_new_block(loopbody);
+        {
+            tmp = builder->CreateAdd(
+                        LLVM::CreateLoad(*builder, i),
+                        llvm::ConstantInt::get(context, llvm::APInt(32, 1)));
+            LLVM::CreateStore(*builder, tmp, i);
+        }
+        builder->CreateBr(loophead);
+
+        // end
+        llvm_utils->start_new_block(loopend);
+
+
+        llvm::Function *fn = builder->GetInsertBlock()->getParent();
+        llvm::BasicBlock *thenBB = llvm::BasicBlock::Create(context, "then", fn);
+        llvm::BasicBlock *elseBB = llvm::BasicBlock::Create(context, "else");
+        llvm::BasicBlock *mergeBB = llvm::BasicBlock::Create(context, "ifcont");
+
+        llvm::Value* cond = builder->CreateICmpEQ(
+                              LLVM::CreateLoad(*builder, i), current_end_point);
+        builder->CreateCondBr(cond, thenBB, elseBB);
+        builder->SetInsertPoint(thenBB);
+        {
+            LCompilersException("The list does not contain the element");
+            // Not working!
+        }
+        builder->CreateBr(mergeBB);
+
+        llvm_utils->start_new_block(elseBB);
+        llvm_utils->start_new_block(mergeBB);
+
+        return LLVM::CreateLoad(*builder, i);
+    }
+
+    void LLVMList::remove(llvm::Value* list, llvm::Value* item) {
+        llvm::Type* pos_type = std::get<2>(typecode2listtype["164"]);
```

Comment:
This line is causing `SEGFAULT`
So, I did this to fix:
```diff
diff --git a/src/libasr/codegen/llvm_utils.cpp b/src/libasr/codegen/llvm_utils.cpp
index b9fb5ebdd..1f43aa37e 100644
--- a/src/libasr/codegen/llvm_utils.cpp
+++ b/src/libasr/codegen/llvm_utils.cpp
@@ -375,7 +375,7 @@ namespace LFortran {
 
     llvm::Value* LLVMList::find_item_position(llvm::Value* list,
                                     llvm::Value* item) {
-        llvm::Type* pos_type = std::get<2>(typecode2listtype["i64"]);
+        llvm::Type* pos_type = llvm::Type::getInt32Ty(context);
         llvm::Value* current_end_point = LLVM::CreateLoad(*builder,
                                         get_pointer_to_current_end_point(list));
         llvm::AllocaInst *i = builder->CreateAlloca(pos_type, nullptr);
@@ -444,7 +444,7 @@ namespace LFortran {
     }
 
     void LLVMList::remove(llvm::Value* list, llvm::Value* item) {
-        llvm::Type* pos_type = std::get<2>(typecode2listtype["164"]);
+        llvm::Type* pos_type = llvm::Type::getInt32Ty(context);
         llvm::Value* current_end_point = LLVM::CreateLoad(*builder,
                                         get_pointer_to_current_end_point(list));
         llvm::AllocaInst *item_pos = builder->CreateAlloca(pos_type, nullptr);
``

Reply:
Yes. We already know the type of position so no need to access it from `typecode2listtype`.